en/ros/ros2_comm.md

```diff
@@ -0,0 +1,579 @@
+# PX4-ROS 2 Bridge
+
+This section of the guide focusses on the bridge between PX4 and ROS 2, made available through the *PX4-Fast RTPS(DDS) Bridge* and resulting products to interface with ROS 2, *px4_ros_com* and *px4_msgs* ROS 2 packages.
+
+It also provides information in how to:
+1. Connect ROS 2 nodes with PX4 (via the *microRTPS* Bridge, and using the `px4_ros_com` package)
+1. Connect ROS (ROS "version 1") nodes with PX4 by additionally using the `ros1_bridge` package to bridge ROS 2 and ROS (1).
+
+:::note
+For information about using the *PX4-Fast RTPS(DDS) Bridge* without ROS 2, see the [RTPS/DDS Interface section](../middleware/micrortps.md).
+:::
+
+:::note
+For a more detailed and visual explanation of using the bridge with ROS 2, see the following talks/presentation videos:
+1. [PX4 Dev Summit 2019 - "ROS 2 Powered PX4"](https://www.youtube.com/watch?v=2Szw8Pk3Z0Q)
+1. [ROS World 2020 - Getting started with ROS 2 and PX4](https://www.youtube.com/watch?v=qhLATrkA_Gw)
+:::
+
+## Why Two Bridges (PX4/ROS2 vs PX4/FastDDS)?
+
+The [PX4-Fast RTPS(DDS) Bridge](../middleware/micrortps.md) and PX4-ROS2 bridge (described here) are conceptually the same.
```

Comment:
@TSC21 This is my attempt to capture the difference between the bridges, as discussed in https://github.com/PX4/PX4-user_guide/pull/1014/files#r569971259
The reason I think this is useful is because you otherwise don't have an overview of the application BUILD process - ie production of px4_msgs.

Please check this CAREFULLY. I am assuming that the client code is the same for both normal and ROS. The IDLs for ROS are generated to px4_msgs, which are built to ROS2 agent code using `px4_ros_com`.

Reply:
> Please check this CAREFULLY. I am assuming that the client code is the same for both normal and ROS.

Yes correct.

> The IDLs for ROS are generated to px4_msgs, which are built to ROS2 agent code using px4_ros_com.

No. px4_msgs generates the IDL codes used by the code generators in the px4_ros_com for the agent.

Reply:
OK. let's be clear. This is what it says now:

> The PX4 build process automatically generates the IDL file-format required for ROS 2 and publishes them to `px4_msgs`.
> The `px4_msgs` are then processed by the agent code generator in `px4_ros_com` to create the agent source code.

You are saying that  `px4_msgs` generates IDL codes NOT PX4 build process. Further that these IDL codes are used by the code generators in the px4_ros_com to create the agent.

What/when does  `px4_msgs` use as the source for generating the IDL codes? I THINK from other discussions that these are essentially generated by the firmware as part of a build. Correct?